File: src/main/java/com/rotpaddon/exampleaddon/network/server/TutorialDataPacket.java
package com.rotpaddon.exampleaddon.network.server;

import java.util.function.Supplier;

import com.github.standobyte.jojo.client.ClientUtil;
import com.github.standobyte.jojo.network.packets.IModPacketHandler;
import com.github.standobyte.jojo.power.impl.nonstand.INonStandPower;
import com.rotpaddon.exampleaddon.init.InitPowers;

import net.minecraft.entity.Entity;
import net.minecraft.entity.LivingEntity;
import net.minecraft.network.PacketBuffer;
import net.minecraftforge.fml.network.NetworkEvent;

public class TutorialDataPacket {
    public final int entityId;
    public final boolean isArmored;

    public TutorialDataPacket(int entityId, boolean isArmored) {
        this.entityId = entityId;
        this.isArmored = isArmored;
    }

    public static class Handler implements IModPacketHandler<TutorialDataPacket> {
        @Override
        public void encode(TutorialDataPacket msg, PacketBuffer buf) {
            buf.writeInt(msg.entityId);
            buf.writeBoolean(msg.isArmored);
        }

        @Override
        public TutorialDataPacket decode(PacketBuffer buf) {
            int entityId = buf.readInt();
            boolean isArmored = buf.readBoolean();
            return new TutorialDataPacket(entityId, isArmored);
        }

        @Override
        public void handle(TutorialDataPacket msg, Supplier<NetworkEvent.Context> ctx) {
            Entity entity = ClientUtil.getEntityById(msg.entityId);
            if (entity instanceof LivingEntity) {
            	INonStandPower.getNonStandPowerOptional((LivingEntity) entity).ifPresent(power -> {
	                power.getTypeSpecificData(InitPowers.TUTORIAL.get()).ifPresent(data -> {
	                    data.setIsArmored(msg.isArmored);
	                });
            	});
            }
        }

        @Override
        public Class<TutorialDataPacket> getPacketClass() {
            return TutorialDataPacket.class;
        }
    }
}

File: src/main/kotlin/com/rotpaddon/exampleaddon/network/server/TutorialDataPacket.kt
package com.rotpaddon.exampleaddon.network.server

import com.github.standobyte.jojo.client.ClientUtil
import com.github.standobyte.jojo.network.packets.IModPacketHandler
import com.github.standobyte.jojo.power.impl.nonstand.INonStandPower
import com.rotpaddon.exampleaddon.init.InitPowers
import com.rotpaddon.exampleaddon.power.impl.nonstand.type.tutorial.TutorialData
import com.rotpaddon.exampleaddon.power.impl.nonstand.type.tutorial.TutorialPowerType
import net.minecraft.entity.LivingEntity
import net.minecraft.network.PacketBuffer
import net.minecraftforge.fml.network.NetworkEvent
import java.util.function.Supplier

class TutorialDataPacket(val entityId: Int, val isArmored: Boolean) {

    class Handler : IModPacketHandler<TutorialDataPacket> {
        override fun encode(msg: TutorialDataPacket, buf: PacketBuffer) {
            buf.writeInt(msg.entityId)
            buf.writeBoolean(msg.isArmored)
        }

        override fun decode(buf: PacketBuffer): TutorialDataPacket {
            val entityId = buf.readInt()
            val isArmored = buf.readBoolean()
            return TutorialDataPacket(entityId, isArmored)
        }

        override fun handle(msg: TutorialDataPacket, ctx: Supplier<NetworkEvent.Context>) {
            val entity = ClientUtil.getEntityById(msg.entityId)
            if (entity is LivingEntity) {
                INonStandPower.getNonStandPowerOptional(entity).ifPresent { power ->
                    power.getTypeSpecificData<TutorialPowerType, TutorialData>(InitPowers.TUTORIAL.get()).ifPresent { data ->
                        data.setIsArmored(msg.isArmored)
                    }
                }
            }
        }

        override fun getPacketClass(): Class<TutorialDataPacket> {
            return TutorialDataPacket::class.java
        }
    }
}